rust/README.md

```diff
@@ -0,0 +1,135 @@
+# Practical Networked Applications in Rust
+
+A training course on practical software construction in Rust.
+
+The final project in this course is a networked, persistent [key-value store]
+with multithreading, asynchronous I/O (via [tokio] and [tower]), a
+high-performance network protocol ([GRPC] via [prost] and [tower-grpc]),
+backed by a simple [LSM-tree] that the student will write themselves.
```

Comment:
oh, building an LSM tree is interesting but hard work.
I think this is also a challenge for us to teach students something about how to use Rust to write this :-)

Reply:
Agreed it's hard, and I'm on the fence about it. It would definitely be an "extra credit" type thing. And I'm not necessarily thinking a full LSM-tree. More like "when the WAL hits a threshold, dump it to a binary-searchable file; when too many files accumulate merge them", but even that is quite a lot of work. The thing I like about it is that it exercises file I/O apis more than simple serialization and that it exercises an algorithm using those APIs.

I'll just remove this wording, maybe leave hints of it in the lesson plan, and see as we go if it's just too hard to ask students to think about.